src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenCrossVersionTest.groovy

```diff
@@ -0,0 +1,47 @@
+package hudson.plugins.gradle.injection
+
+
+import hudson.plugins.gradle.BuildScanAction
+import hudson.plugins.gradle.BuildScanBuildWrapper
+import hudson.tasks.Maven
+import org.jvnet.hudson.test.CreateFileBuilder
+import org.jvnet.hudson.test.JenkinsRule
+import org.jvnet.hudson.test.ToolInstallations
+import spock.lang.Unroll
+
+@Unroll
+class BuildScanInjectionMavenCrossVersionTest extends BaseInjectionIntegrationTest {
+
+    def 'build scan is discovered from Maven build - #mavenVersion'(String mavenVersion) {
+        given:
+        withGlobalEnvVars {
+            put('JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION', 'true')
+            put('JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION', '1.14.2')
+            put('JENKINSGRADLEPLUGIN_CCUD_EXTENSION_VERSION', '1.10.1')
+        }
+
+        def p = j.createFreeStyleProject()
+        p.buildWrappersList.add(new BuildScanBuildWrapper())
+        p.buildersList.add(new CreateFileBuilder('pom.xml', MavenExamples.simplePom()))
+        p.buildersList.add(new CreateFileBuilder('.mvn/gradle-enterprise.xml', MavenExamples.gradleEnterpriseConfiguration()))
+        def mavenInstallation = ToolInstallations.configureDefaultMaven("apache-maven-${mavenVersion}", Maven.MavenInstallation.MAVEN_30)
```

Comment:
❓ Why not use existing Jenkins download mechanisms, similar to how it's done in Gradle tests?

Reply:
Good ideea, done.